sysbuild/Kconfig.sdp

```diff
@@ -15,6 +15,9 @@ config SDP_GPIO
 config SDP_MSPI
 	bool "SDP MSPI application"
 
+config SDP_DEVELOPER_MODE
+	bool "SDP developer mode"
```

Comment:
should be a choice for GPIO/MSPI, ~~and possible developer? Not sure what developer is~~

Reply:
I am not sure I understand this comment, `SDP_DEVELOPER_MODE` is an indication, what should happen if provided and generated ASM files differ. It should be an option for all SDPs, even those developed by our clients, not only for GPIO and MSPI. Also, it should be disabled by default for all SDPs and it is.
I can make it a Kconfig choice with "developer" and "user" mode as options, but I am not sure if it is worth it, since there are only two options.

Reply:
comment is about the GPIO and MSPI selections, they select a different application so it should be a choice

Reply:
Ok, I understand now. I changed the selection of GPIO/MSPI to a choice, but I put it in a separate commit, since it is not really related to adding developer mode.